code/weapon/weapons.cpp

```diff
@@ -1480,7 +1480,30 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		}
 	}
 
-
+		// set the number of missles or ballistic ammo reloaded per batch
+	int reloaded_each_batch;
+	
+	if (optional_string("$Rearm Ammo Increment:")) {
+		stuff_int(&reloaded_each_batch);
+
+		if (reloaded_each_batch > 0) {
+			wip->reloaded_per_batch = reloaded_each_batch;
+		} else if (wip->reloaded_per_batch != -1) {
+			Warning(LOCATION, "Rearm increment amount of 0 or less in weapon %s, keeping at previously defined value.",
+			        wip->name);
+		} else {
+			Warning(LOCATION, "Rearm increment amount of 0 or less in weapon %s, setting to the default, 4 for missiles or 100 for ballistic primaries.", wip->name);
+		}
+	}
+	// This setting requires a default if not specified, otherwise, rearming will break. - Cyborg17
+	if (wip->reloaded_per_batch == -1) {
```

Comment:
Change this to `else if (first_time)` instead.

Reply:
Nevermind, then you'd have to duplicate code to set the defaults if an invalid value was specified.